main.go

```diff
@@ -18,22 +18,51 @@ func getAdFromBs(w http.ResponseWriter, r *http.Request) {
 	vars := mux.Vars(r)
 	dateString := vars["date"]
 	var splitedDate = strings.Split(dateString, "-")
+	if len(splitedDate) < 3 || len(splitedDate) > 3 {
+		http.Error(w, "not a valid date", http.StatusBadRequest)
+		return
+	} else {
+		day, _ := strconv.Atoi(splitedDate[2])
+		month, _ := strconv.Atoi(splitedDate[1])
+		year, _ := strconv.Atoi(splitedDate[0])
+		date, err := bsdate.New(day, month, year)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+		gregorianDate, _ := date.GetGregorianDate()
+		fmt.Fprintf(w, gregorianDate.Format("2006-01-02"))
+	}
+}
+
+func getBsFromAd(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	dateString := vars["date"]
+	var splitedDate = strings.Split(dateString, "-")
+	if len(splitedDate) < 3 || len(splitedDate) > 3 {
+			http.Error(w, "cannot convert date, missing data", http.StatusBadRequest)
```

Comment:
wrong indent
if the amount of fields is not correct, is that then missing data? e.g. if the input is `1-2-3-4`

Reply:
yes I thought of taking those as invalid data. What can be done better?